### PR TITLE
Connecting Semesters to Profile

### LINKED ISSUE #357: [Internal] Connect Starting & Ending Semesters to Profile
Currently, the "Profile" section lists name, email, major, and minor. However, we no longer ask for major and minor in onboarding; instead we ask for starting and ending semester. The profile page should reflect that and allow the user to change their starting and ending semester.

Screenshot 2023-01-30 at 8 48 11 AM

Note: THIS WILL BE WORKED ON AS AN ONBOARDING TASK DURING THE UPCOMING PLANNER MEETING. **DO NOT** WORK ON THIS, YOUR PR WILL BE CLOSED

### PR DESCRIPTION
**Pull request by Peytonbarre**
## Overview
Resolves #357 
Changed the profile page to include the semesters and removed the majors and minors option.

## What Changed
Changed the profile component to remove the majors and minors text fields. Also changed the handle submit to include starting and ending semesters. Changed the user trpc updateUserProfile to allow these changes.

## Other Notes
Had a few problems with non-null assertions. Removed the ones giving in error in useEffect but kept them in for the handleSubmit. Don't know if using non-null assertions is good practice or not, I can change them if needed.

### CODE REVIEW

File: src/components/home/Profile.tsx [hunk 1]
```diff
@@ -39,10 +65,62 @@ export default function ProfilePage({ isDesktop }: ProfilePageProps): JSX.Elemen
...
+        firstSemester = 'f';
+        break;
+      case "Spring":
+        firstSemester = 's';
+        break;
+      case "Summer":
+        firstSemester = 'u';
+        break;
+    }
+    console.log(firstSem)
```
Comment (thekevinge) [inline src/components/home/Profile.tsx:82]: Don't forget to delete the console logs :)

**Review by akevinge — CHANGES_REQUESTED** [review]
A few related things (no changes needed for these):
- Currently/planning on migrating off MUI 
- Users shouldn't be able to make their end semester before their start semester, but that'll be in another issue 

Overall good stuff! Great job :tada:

**Comment by thekevinge** [comment]
Oh and also make sure your code is formatted, just noticed the workflow failed. You can run ``npx  prettier -w .`` from the project root or install the VSCode extension (if you use that).

**Review by ZzRanger — DISMISSED** [review]
It works good job! Congrats on first issue for the semester :)

File: src/components/home/Profile.tsx [hunk 2]
```diff
@@ -39,10 +65,62 @@ export default function ProfilePage({ isDesktop }: ProfilePageProps): JSX.Elemen
 
   const handleSubmit = () => {
     // TODO: Implement changing profile settings here
```
[the remark this answers is not in the document]
Comment (ZzRanger) [reply src/components/home/Profile.tsx:81]: Oh uhh do that real quick then your good to go

**Pull request merged**